### PR TITLE
fix: Remove click listeners to remove memory leaks

### PR DESCRIPTION
**Pull request by anhanh11001**
Detail:
- The cause of the memory leak is because of the adapter of RecyclerView. If the adapter lives any longer than the RecyclerView does, the adapter is going to hold a reference to the RecyclerView which should have already gone out of memory.

Further description: https://stackoverflow.com/questions/35520946/leak-canary-recyclerview-leaking-madapter. 

To be honest, I don't completely understand the problem and the memory leak in here, but this fix does really improve the app as it is faster in my device and leak canary doesn't find any leaks. So please correct me if there is anything wrong or give me further explanations.

Fixes: #1780

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt [hunk 1]
```diff
@@ -53,11 +52,14 @@ class EventViewHolder(override val containerView: View) : RecyclerView.ViewHolde
         val startsAt = EventUtils.getEventDateTime(event.startsAt, event.timezone)
         val endsAt = EventUtils.getEventDateTime(event.endsAt, event.timezone)
 
+        containerView.shareFab.scaleType = CENTER
+        containerView.favoriteFab.scaleType = CENTER
```
Comment (ShridharGoel) [inline app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt]: How is this related with the issue?

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/event/EventViewHolder.kt]: Oops. It's not. I just forgot to revert this

**Comment by iamareebjamal** [comment]
Rather than doing this, just remove koin scoping please

**Comment by anhanh11001** [comment]
I will remove Koin Scope. But should I keep the part where setting recyclerView adapter equals to null in other fragments in onDestroyView()? Leak Canary also find leaks in other fragments with the same adapter-recyclerView problem mentioned above.

**Comment by iamareebjamal** [comment]
Honestly, that's a workaround, not a fix. We should try to find out the root cause of the leaks and fix it for the non koin adapters where leak canary is showing leaks. It may be because of nested viewholders or click listeners in adapters

**Comment by anhanh11001** [comment]
@iamareebjamal please take the look at this article as it faces the same problem as we are and please give me some advice on which solution should I go for

https://medium.com/@yfujiki/tracing-simple-memory-leak-around-recyclerview-using-leakcanary-927460532d53

**Comment by iamareebjamal** [comment]
I'll take a look over the weekend.

**Comment by anhanh11001** [comment]
@iamareebjamal Have you taken a look at the blog post?

**Comment by iamareebjamal** [comment]
Yes, the blog post is simply telling the problem. I'll analyse the app today to tell what can be the fix

**Comment by iamareebjamal** [comment]
Memory leaks are due to click listeners. Please remove them onDestroyView

**Comment by anhanh11001** [comment]
updated
